From 3a9d9a892baa19809671b66ff543fada930dbf49 Mon Sep 17 00:00:00 2001
From: Pierre Joye <pierre.php@gmail.com>
Date: Sat, 29 Jan 2022 23:18:28 +0700
Subject: [PATCH 22/40] Fix #810, Wrong image freed, src_cloned should be freed
 if set. (#811)

---
 src/gd_interpolation.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gd_interpolation.c b/src/gd_interpolation.c
index d1ebfc9c..11c24469 100644
--- a/src/gd_interpolation.c
+++ b/src/gd_interpolation.c
@@ -1857,7 +1857,7 @@ BGD_DECLARE(gdImagePtr) gdImageRotateInterpolated(const gdImagePtr src, const fl
 
 		case -18000:
 		case  18000:
-			if (src_cloned) gdImageDestroy(src);
+			if (src_cloned) gdImageDestroy(src_tc);
 			return gdImageRotate180(src, 0);
 
 		case  -9000:
-- 
2.40.1

