diff -up libgd-2.3.3/src/gd_bmp.c.omv~ libgd-2.3.3/src/gd_bmp.c
--- libgd-2.3.3/src/gd_bmp.c.omv~	2023-04-28 17:57:25.067641910 +0200
+++ libgd-2.3.3/src/gd_bmp.c	2023-04-28 17:58:35.989307122 +0200
@@ -678,6 +678,8 @@ static int bmp_read_windows_v3_info(gdIO
 
 static int bmp_read_os2_v1_info(gdIOCtxPtr infile, bmp_info_t *info)
 {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
 	if (
 	    !gdGetWordLSB((signed short int *)&info->width, infile) ||
 	    !gdGetWordLSB((signed short int *)&info->height, infile) ||
@@ -686,6 +688,7 @@ static int bmp_read_os2_v1_info(gdIOCtxP
 	) {
 		return 1;
 	}
+#pragma GCC diagnostic pop
 
 	/* OS2 v1 doesn't support topdown */
 	info->topdown = 0;
diff -up libgd-2.3.3/src/gd_nnquant.c.omv~ libgd-2.3.3/src/gd_nnquant.c
--- libgd-2.3.3/src/gd_nnquant.c.omv~	2023-04-28 17:54:10.147822658 +0200
+++ libgd-2.3.3/src/gd_nnquant.c	2023-04-28 17:56:41.571234713 +0200
@@ -112,12 +112,7 @@ typedef struct {
 
 /* Initialise network in range (0,0,0,0) to (255,255,255,255) and set parameters
    ----------------------------------------------------------------------- */
-static void initnet(nnq, thepic, len, sample, colours)
-nn_quant *nnq;
-unsigned char *thepic;
-int len;
-int sample;
-int colours;
+static void initnet(nn_quant *nnq, unsigned char *thepic, int len, int sample, int colours)
 {
 	register int i;
 	register int *p;
@@ -163,9 +158,7 @@ static void unbiasnet(nn_quant *nnq)
 }
 
 /* Output colormap to unsigned char ptr in RGBA format */
-static void getcolormap(nnq, map)
-nn_quant *nnq;
-unsigned char *map;
+static void getcolormap(nn_quant *nnq, unsigned char *map)
 {
 	int i,j;
 	for(j=0; j < nnq->netsize; j++) {
@@ -232,9 +225,7 @@ static void inxbuild(nn_quant *nnq)
 
 /* Search for ABGR values 0..255 (after net is unbiased) and return colour index
 	 ---------------------------------------------------------------------------- */
-static unsigned int inxsearch(nnq, al,b,g,r)
-nn_quant *nnq;
-register int al, b, g, r;
+static unsigned int inxsearch(nn_quant *nnq, int al, int b, int g, int r)
 {
 	register int i, j, dist, a, bestd;
 	register int *p;
@@ -306,9 +297,7 @@ register int al, b, g, r;
 
 /* Search for biased ABGR values
    ---------------------------- */
-static int contest(nnq, al,b,g,r)
-nn_quant *nnq;
-register int al,b,g,r;
+static int contest(nn_quant *nnq, int al, int b, int g, int r)
 {
 	/* finds closest neuron (min dist) and updates freq */
 	/* finds best neuron (min dist-bias) and returns position */
@@ -362,9 +351,7 @@ register int al,b,g,r;
 /* Move neuron i towards biased (a,b,g,r) by factor alpha
 	 ---------------------------------------------------- */
 
-static void altersingle(nnq, alpha,i,al,b,g,r)
-nn_quant *nnq;
-register int alpha,i,al,b,g,r;
+static void altersingle(nn_quant *nnq, int alpha, int i, int al, int b, int g, int r)
 {
 	register int *n;
 
@@ -382,10 +369,7 @@ register int alpha,i,al,b,g,r;
 /* Move adjacent neurons by precomputed alpha*(1-((i-j)^2/[r]^2)) in radpower[|i-j|]
 	 --------------------------------------------------------------------------------- */
 
-static void alterneigh(nnq, rad,i,al,b,g,r)
-nn_quant *nnq;
-int rad,i;
-register int al,b,g,r;
+static void alterneigh(nn_quant *nnq, int rad, int i, int al, int b, int g, int r)
 {
 	register int j,k,lo,hi,a;
 	register int *p, *q;
@@ -429,9 +413,7 @@ register int al,b,g,r;
 /* Main Learning Loop
    ------------------ */
 
-static void learn(nnq, verbose) /* Stu: N.B. added parameter so that main() could control verbosity. */
-nn_quant *nnq;
-int verbose;
+static void learn(nn_quant *nnq, int verbose) /* Stu: N.B. added parameter so that main() could control verbosity. */
 {
 	register int i,j,al,b,g,r;
 	int radius,rad,alpha,step,delta,samplepixels;
diff -up libgd-2.3.3/src/gd_tiff.c.omv~ libgd-2.3.3/src/gd_tiff.c
--- libgd-2.3.3/src/gd_tiff.c.omv~	2023-04-28 17:56:48.179296534 +0200
+++ libgd-2.3.3/src/gd_tiff.c	2023-04-28 17:57:04.364448016 +0200
@@ -446,9 +446,7 @@ BGD_DECLARE(void) gdImageTiffCtx(gdImage
 }
 
 /* Check if we are really in 8bit mode */
-static int checkColorMap(n, r, g, b)
-int n;
-uint16_t *r, *g, *b;
+static int checkColorMap(int n, uint16_t *r, uint16_t *g, uint16_t *b)
 {
 	while (n-- > 0)
 		if (*r++ >= 256 || *g++ >= 256 || *b++ >= 256)
diff -up libgd-2.3.3/src/gd_tiff.c.omv~ libgd-2.3.3/src/gd_tiff.c
--- libgd-2.3.3/src/gd_tiff.c.omv~	2023-04-28 19:17:54.596160900 +0200
+++ libgd-2.3.3/src/gd_tiff.c	2023-04-28 19:20:31.667599629 +0200
@@ -951,8 +951,18 @@ BGD_DECLARE(gdImagePtr) gdImageCreateFro
 		image_type = GD_RGB;
 	}
 
+// False positive with gcc 13.1
+// It thinks image_type can be uninitialized, but that
+// can't actually happen
+#ifndef __clang__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#endif
 	if (!force_rgba &&
 	        (image_type == GD_PALETTE || image_type == GD_INDEXED || image_type == GD_GRAY)) {
+#ifndef __clang__
+#pragma GCC diagnostic pop
+#endif
 		im = gdImageCreate(width, height);
 		if (!im) goto error;
 		readTiffColorMap(im, tif, is_bw, photometric);
